gremlin-go/driver/driverRemoteConnection.go

```diff
@@ -181,10 +181,28 @@ func (driver *DriverRemoteConnection) CreateSession(sessionId ...string) (*Drive
 	drc, err := NewDriverRemoteConnection(driver.client.url, func(settings *DriverRemoteConnectionSettings) {
 		settings.TraversalSource = driver.client.traversalSource
 		if len(sessionId) == 1 {
-			settings.Session = sessionId[0]
+			settings.session = sessionId[0]
 		} else {
-			settings.Session = uuid.New().String()
+			settings.session = uuid.New().String()
 		}
+
+		settings.TraversalSource = driver.client.traversalSource
+		settings.TransporterType = driver.client.transporterType
+
+		settings.Logger = driver.client.logHandler.logger
+		settings.LogVerbosity = driver.client.logHandler.verbosity
+		//settings.Language = driver.client.logHandler
+
+		settings.AuthInfo = driver.client.connections.(*loadBalancingPool).connSettings.authInfo
+		settings.TlsConfig = driver.client.connections.(*loadBalancingPool).connSettings.tlsConfig
+		settings.KeepAliveInterval = driver.client.connections.(*loadBalancingPool).connSettings.keepAliveInterval
+		settings.WriteDeadline = driver.client.connections.(*loadBalancingPool).connSettings.writeDeadline
+		settings.ConnectionTimeout = driver.client.connections.(*loadBalancingPool).connSettings.connectionTimeout
+		settings.NewConnectionThreshold = driver.client.connections.(*loadBalancingPool).newConnectionThreshold
+		settings.EnableCompression = driver.client.connections.(*loadBalancingPool).connSettings.enableCompression
+		settings.ReadBufferSize = driver.client.connections.(*loadBalancingPool).connSettings.readBufferSize
+		settings.WriteBufferSize = driver.client.connections.(*loadBalancingPool).connSettings.writeBufferSize
+		settings.MaximumConcurrentConnections = cap(driver.client.connections.(*loadBalancingPool).connections)
```

Comment:
Instead of all this which will very easily break, I think you should just make NewDriverRemoteConnection keep a copy of all the settings when it's consutructed so you can reference it again here.